x-pack/plugins/ingest_manager/server/services/epm/packages/install.ts

```diff
@@ -6,22 +6,69 @@
 
 import { SavedObject, SavedObjectsClientContract } from 'src/core/server/';
 import { PACKAGES_SAVED_OBJECT_TYPE } from '../../../constants';
-import { AssetReference, Installation, KibanaAssetType, CallESAsCurrentUser } from '../../../types';
+import {
+  AssetReference,
+  Installation,
+  KibanaAssetType,
+  CallESAsCurrentUser,
+  DefaultPackages,
+} from '../../../types';
 import { installIndexPatterns } from '../kibana/index_pattern/install';
 import * as Registry from '../registry';
 import { getObject } from './get_objects';
-import { getInstallation } from './index';
+import { getInstallation, findInstalledPackageByName } from './index';
 import { installTemplates } from '../elasticsearch/template/install';
 import { installPipelines } from '../elasticsearch/ingest_pipeline/install';
 import { installILMPolicy } from '../elasticsearch/ilm/install';
 
+export async function installLatestPackage(options: {
+  savedObjectsClient: SavedObjectsClientContract;
+  pkgName: string;
+  callCluster: CallESAsCurrentUser;
+}): Promise<AssetReference[]> {
+  const { savedObjectsClient, pkgName, callCluster } = options;
+  try {
+    const latestPackage = await Registry.fetchFindLatestPackage(pkgName);
+    const pkgkey = Registry.pkgToPkgKey({
+      name: latestPackage.name,
+      version: latestPackage.version,
+    });
+    return installPackage({ savedObjectsClient, pkgkey, callCluster });
+  } catch (err) {
+    throw err;
+  }
+}
+
+export async function ensureInstalledDefaultPackages(
+  savedObjectsClient: SavedObjectsClientContract,
+  callCluster: CallESAsCurrentUser
+): Promise<void> {
+  for (const pkgName in DefaultPackages) {
+    if (!DefaultPackages.hasOwnProperty(pkgName)) continue;
+    const installedPackage = await findInstalledPackageByName({ savedObjectsClient, pkgName });
+    // if the requested packaged was not found to be installed, try installing
+    if (!installedPackage) {
+      try {
+        await installLatestPackage({
```

Comment:
@ruflin if a user uninstalls the `system` package, should we be installing it every time? Or will they not be allowed to uninstall `system` via the UI, only "disable" it?

Reply:
They should not be able to uninstall the system package. What would mean disable?

Reply:
Thanks.  Wasn't sure if we planned to have some special state planned for it after looking at the design where you can "enable system monitoring" when create a new agent configuration.

Reply:
On the package side, we should probably just not show the uninstall button and have a message. @hbharding